packages/bitcore-node/test/util/verify.ts

```diff
@@ -0,0 +1,232 @@
+  expect(heights.filter(h => !h).length, 'no duplicate heights').to.equal(0);
+
+  // Check increasing times
+  const increases = l => !!l.reduce((prev, curr) => prev < curr? curr : undefined);
+  expect(increases(times), 'block times only increase').to.be.true;
+  expect(increases(normalizedTimes), 'normalized block times only increase').to.be.true;
+}
+
+
+export async function transactions(info: ChainNetwork, creds: {
+  username: string;
+  password: string;
+  host: string;
+  port: number;
+}) {
+  const rpc = new AsyncRPC(creds.username, creds.password, creds.host, creds.port);
+
+  const txcursor = TransactionModel.collection.find({
+    chain: info.chain,
+    network: info.network,
+  });
+
+  while (true) {
+    const tx: ITransaction = await txcursor.next();
+    if (!tx) {
+      break;
+    }
+    logger.info(`verifying tx ${tx.txid}: ${tx.blockHeight}`);
+    const truth = await rpc.transaction(tx.txid, tx.blockHash);
+
+    expect(tx.size, 'tx size').to.equal(truth.size);
+    expect(tx.locktime, 'tx locktime').to.equal(truth.locktime);
+
+    { // Minted by this transaction
+      const ours = await CoinModel.collection.find({
+        network: info.network,
+        chain: info.chain,
+        mintTxid: tx.txid,
+      }).toArray();
+      expect(ours.length, 'number mint txids').to.equal(truth.vout.length);
+      for (const our of ours) {
+        // coins
+        expect(our.mintHeight, 'tx mint height').to.equal(tx.blockHeight);
+        expect(our.value, 'tx mint value').to.equal(
+          Math.round(truth.vout[our.mintIndex].value * SATOSHI)
+        );
+        // TODO: why?
+        if (our.address && our.address !== 'false') {
+          expect(truth.vout[our.mintIndex].scriptPubKey.addresses,
+                 'tx mint address').to.include(our.address);
+        }
+        expect(our.coinbase).to.equal(tx.coinbase);
+
+        // wallets
+        expect(tx.wallets).to.include.members(Array.from(our.wallets));
+        if (our.wallets.size > 0) {
+          const wallets = await WalletAddressModel.collection.find({
+            wallet: {
+              $in: our.wallets,
+            },
+            address: our.address,
+            chain: info.chain,
+            network: info.network,
+          }).toArray();
+          expect(wallets.length, 'wallet exists').to.be.greaterThan(0);
+        }
+      }
+    }
+
+    { // Spent by this transaction
+      const ours = await CoinModel.collection.find({
+        network: info.network,
+        chain: info.chain,
+        spentTxid: tx.txid,
+      }).toArray();
+      const nspent = truth.vin.length + (tx.coinbase ? -1 : 0);
+      expect(ours.length, 'number spent txids').to.equal(nspent);
+      for (const our of ours) {
+        expect(our.spentHeight, 'tx spent height').to.equal(tx.blockHeight);
+        expect(tx.wallets).to.include.members(Array.from(our.wallets));
+      }
+    }
+  }
+}
+
+
+if (require.main === module) (async () => {
```

Comment:
This looks weird

Reply:
nah it's good

Reply:
https://nodejs.org/docs/latest/api/all.html#modules_accessing_the_main_module

Reply:
So it's something that is exported (so it can be required), but if it is called directly it runs as a script

Reply:
To accomplish the same thing you could move the logic to another file that exports and have the scripts just require and run the file